app/routes.py

```diff
@@ -1,2 +1,277 @@
+    
+    task = Task.query.get_or_404(task_id)
+    task.completed_at = None
+
+    db.session.commit()
+
+    return jsonify({
+            "task": {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": False if task.completed_at == None else True
+            }
+        }), 200
+
+from app.models.goal import Goal
+
+goals_bp = Blueprint("goals", __name__, url_prefix = "/goals")
+
+@goals_bp.route("", methods = ["POST", "GET"])
+def handle_goals():
+    if request.method == "POST":
+    
+        request_body = request.get_json()
+
+        if "title" not in request_body:
+            return jsonify({
+                "details": "Invalid data"
+            }), 400
+        else:
+            new_goal = Goal(title = request_body["title"])
+            db.session.add(new_goal)
+            db.session.commit()
+            
+        return jsonify({
+                "goal": {
+                    "id": new_goal.goal_id,
+                    "title": new_goal.title
+                }
+            }), 201
+
+    elif request.method == "GET":
+        goal_query = request.args.get("title")
+
+        if goal_query:
+            goals = Goal.query.filter_by(title = goal_query)
+        else:
+            goals = Goal.query.all()
+        
+        list_of_goals = []
+        for goal in goals:
+            list_of_goals.append({
+                "id": goal.goal_id,
+                "title": goal.title
+            })
+
+        if list_of_goals is None:
+            return jsonify([]), 200
+        else:
+            return jsonify(list_of_goals), 200
+
+@goals_bp.route("/<goal_id>", methods = ["GET", "PUT", "DELETE"])
+def handle_specific_goals(goal_id):
+    goal = Goal.query.get_or_404(goal_id)
+    
+
+    if goal is None:
+        return jsonify(None), 404
+
+    if request.method == "GET":
+        return {
+            "goal": {
+            "id": goal.goal_id,
+            "title": goal.title
+        }}, 200
+
+    elif request.method == "PUT":
+        updated_body = request.get_json()
+
+        goal.title = updated_body["title"]
+        db.session.commit()
+
+        return jsonify({
+            "goal": {
+            "id": goal.goal_id,
+            "title": goal.title
+        }}), 200
+
+    elif request.method == "DELETE":
+        db.session.delete(goal)
+        db.session.commit()
+
+        return jsonify({
+            "details": f'Goal {goal.goal_id} "{goal.title}" successfully deleted'
+        }), 200
+        
+
+@goals_bp.route("/<goal_id>/tasks", methods = ["POST", "GET"])
+def handle_tasks_in_goals(goal_id):
+    goal = Goal.query.get(goal_id)
+
+    if goal is None:
+        return jsonify(None), 404
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        task_ids = request_body["task_ids"]
+
+        for task_id in task_ids:
+            task = Task.query.get(task_id)
```

Comment:
What would happen if any of the tasks we looked up here were invalid ids (or not numeric ids at all)? We should make sure that the task ids actually load a valid task, otherwise the following line will try to access a `NoneType` object.